src/theme/themes/utils/generate-theme.scss

```diff
@@ -1,16 +1,36 @@
 @use "sass:color";
+@use "sass:map";
 @use "./contrast-colour.scss";
 
-@function getColorPalette($primary, $secondary) {
+// Returns a map of colours based partly on Material Design's colour system
+// https://material.io/design/color/the-color-system.html
+@function getColorPalette($primary, $secondary, $page-background: null) {
   $result: (
     "color-primary": $primary,
-    "color-primary-light": color.change($primary, $lightness: 80%),
-    "color-primary-lighter": color.change($primary, $lightness: 95%),
+    "color-primary-50": color.change($primary, $lightness: 95%),
+    "color-primary-100": color.change($primary, $lightness: 90%),
+    "color-primary-200": color.change($primary, $lightness: 80%),
+    "color-primary-300": color.change($primary, $lightness: 70%),
+    "color-primary-400": color.change($primary, $lightness: 60%),
+    "color-primary-500": color.change($primary, $lightness: 50%),
+    "color-primary-600": color.change($primary, $lightness: 40%),
+    "color-primary-700": color.change($primary, $lightness: 30%),
+    "color-primary-800": color.change($primary, $lightness: 20%),
+    "color-primary-900": color.change($primary, $lightness: 10%),
     "color-primary-contrast": contrast-colour.choose-contrast-color($primary),
     "color-secondary": $secondary,
-    "color-secondary-light": color.change($secondary, $lightness: 80%),
-    "color-secondary-lighter": color.change($secondary, $lightness: 95%),
+    "color-secondary-50": color.change($secondary, $lightness: 95%),
```

Comment:
nit(non-blocking) I'm guessing there's a way to generate the 50-900 given the input primary/secondary to halve this code, but no biggie while we're only working with two base colours

Reply:
See example link alongside comment below (likely can reduce code amount significantly, but again no urgency to do)
https://stackoverflow.com/questions/40221097/sass-how-to-generate-a-map-using-a-loop